File: Data Structures & Algorithms/subsets-ii/submission-0.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number[][]}
     */
    subsetsWithDup(nums) {
        /**
         * @param {number[][]} acc
         * @param {number[]} cur
         * @param {number} i
         */
        const dfs = (acc, cur, i) => {
            if (i >= nums.length) return [...acc, cur]

            // Left
            acc = dfs(acc, [...cur, nums[i]], i + 1)

            // Right
            let next_i = i + 1
            while (next_i < nums.length && nums[next_i - 1] === nums[next_i])
                next_i += 1
            acc = dfs(acc, [...cur], next_i)

            return acc
        }

        nums.sort((a, b) => a - b)
        return dfs([], [], 0)
    }
}

File: Data Structures & Algorithms/maximum-subarray/submission-0.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        res = None
        for l in range(len(nums)):
            cur = 0
            for r in range(l, len(nums)):
                cur += nums[r]
                if res is None:
                    res = cur
                res = max(res, cur)
        return res
